11 juin 2018 07:54 "Rotem Yaari" <notifications@github.com> a écrit:
> @gpernot I can't merge this since the Travis build is failing. Can you take a look?

travis fails on :
./mongomock/__init__.py:79:1: E402 module level import not at top of file

and i haven't modified __init__.py at all

maybe you have more experience than me on this ?

--- a/Missing_Features.rst
+++ b/Missing_Features.rst
@@ -6,4 +6,5 @@ If i miss to include a feature in the below list, Please feel free to add to the
 
 * $rename operator  - https://docs.mongodb.com/manual/reference/operator/update/rename/
 * create_collection options - https://docs.mongodb.com/v3.2/reference/method/db.createCollection/#definition
-
+* bypass_document_validation options
+* session options

--- a/mongomock/collection.py
+++ b/mongomock/collection.py
@@ -325,11 +325,11 @@ class Collection(object):
         validate_write_concern_params(**kwargs)
         return self._insert(data)
 
-    def insert_one(self, document, bypass_document_validation=False, session=None):
+    def insert_one(self, document, session=None):
         validate_is_mutable_mapping('document', document)
         return InsertOneResult(self._insert(document), acknowledged=True)
 
-    def insert_many(self, documents, ordered=True, bypass_document_validation=False, session=None):
+    def insert_many(self, documents, ordered=True, session=None):
         if not isinstance(documents, collections.Iterable) or not documents:
             raise TypeError('documents must be a non-empty list')
         for document in documents:
@@ -395,21 +395,18 @@ class Collection(object):
             sub_doc = sub_doc[part]
         del sub_doc[key_parts[-1]]
 
-    def update_one(self, filter, update, upsert=False, bypass_document_validation=False,
-                   collation=None, array_filters=None, session=None):
+    def update_one(self, filter, update, upsert=False, session=None):
         validate_ok_for_update(update)
         return UpdateResult(self._update(filter, update, upsert=upsert),
                             acknowledged=True)
 
-    def update_many(self, filter, update, upsert=False, bypass_document_validation=False,
-                    collation=None, session=None):
+    def update_many(self, filter, update, upsert=False, session=None):
         validate_ok_for_update(update)
         return UpdateResult(self._update(filter, update, upsert=upsert,
                                          multi=True),
                             acknowledged=True)
 
-    def replace_one(self, filter, replacement, upsert=False,
-                    bypass_document_validation=False, collation=None, session=None):
+    def replace_one(self, filter, replacement, upsert=False, session=None):
         validate_ok_for_replace(replacement)
         return UpdateResult(self._update(filter, replacement, upsert=upsert),
                             acknowledged=True)
@@ -1096,11 +1093,11 @@ class Collection(object):
                          manipulate, check_keys=True, **kwargs)
             return to_save.get("_id", None)
 
-    def delete_one(self, filter, collation=None, session=None):
+    def delete_one(self, filter, session=None):
         validate_is_mapping('filter', filter)
         return DeleteResult(self._delete(filter), True)
 
-    def delete_many(self, filter, collation=None, session=None):
+    def delete_many(self, filter, session=None):
         validate_is_mapping('filter', filter)
         return DeleteResult(self._delete(filter, multi=True), True)
 

--- a/mongomock/collection.py
+++ b/mongomock/collection.py
@@ -325,11 +325,11 @@ class Collection(object):
         validate_write_concern_params(**kwargs)
         return self._insert(data)
 
-    def insert_one(self, document):
+    def insert_one(self, document, bypass_document_validation=False, session=None):
         validate_is_mutable_mapping('document', document)
         return InsertOneResult(self._insert(document), acknowledged=True)
 
-    def insert_many(self, documents, ordered=True):
+    def insert_many(self, documents, ordered=True, bypass_document_validation=False, session=None):
         if not isinstance(documents, collections.Iterable) or not documents:
             raise TypeError('documents must be a non-empty list')
         for document in documents:
@@ -379,18 +379,37 @@ class Collection(object):
     def _internalize_dict(self, d):
         return {k: copy.deepcopy(v) for k, v in iteritems(d)}
 
-    def update_one(self, filter, update, upsert=False):
+    def _has_key(self, doc, key):
+        key_parts = key.split('.')
+        sub_doc = doc
+        for part in key_parts:
+            if part not in sub_doc:
+                return False
+            sub_doc = sub_doc[part]
+        return True
+
+    def _remove_key(self, doc, key):
+        key_parts = key.split('.')
+        sub_doc = doc
+        for part in key_parts[:-1]:
+            sub_doc = sub_doc[part]
+        del sub_doc[key_parts[-1]]
+
+    def update_one(self, filter, update, upsert=False, bypass_document_validation=False,
+                   collation=None, array_filters=None, session=None):
         validate_ok_for_update(update)
         return UpdateResult(self._update(filter, update, upsert=upsert),
                             acknowledged=True)
 
-    def update_many(self, filter, update, upsert=False):
+    def update_many(self, filter, update, upsert=False, bypass_document_validation=False,
+                    collation=None, session=None):
         validate_ok_for_update(update)
         return UpdateResult(self._update(filter, update, upsert=upsert,
                                          multi=True),
                             acknowledged=True)
 
-    def replace_one(self, filter, replacement, upsert=False):
+    def replace_one(self, filter, replacement, upsert=False,
+                    bypass_document_validation=False, collation=None, session=None):
         validate_ok_for_replace(replacement)
         return UpdateResult(self._update(filter, replacement, upsert=upsert),
                             acknowledged=True)
@@ -1077,11 +1096,11 @@ class Collection(object):
                          manipulate, check_keys=True, **kwargs)
             return to_save.get("_id", None)
 
-    def delete_one(self, filter):
+    def delete_one(self, filter, collation=None, session=None):
         validate_is_mapping('filter', filter)
         return DeleteResult(self._delete(filter), True)
 
-    def delete_many(self, filter):
+    def delete_many(self, filter, collation=None, session=None):
         validate_is_mapping('filter', filter)
         return DeleteResult(self._delete(filter, multi=True), True)
 
@@ -1310,7 +1329,7 @@ class Collection(object):
         ret_array = ret_array_copy
         return ret_array
 
-    def aggregate(self, pipeline, **kwargs):
+    def aggregate(self, pipeline, session=None, **kwargs):
         pipeline_operators = [
             '$project',
             '$match',

--- a/tox.ini
+++ b/tox.ini
@@ -20,6 +20,6 @@ deps = hacking==1.0.0
 commands = flake8 {posargs}
 
 [flake8]
-ignore = H803,H301,H302,H802
+ignore = H803,H301,H302,H802,E402
 exclude = .venv,.tox,dist,doc,*.egg,build,.ropeproject
 max-line-length = 100

